### ISSUE TITLE
fields that start with a capital can't be found by hash key unless specified.

### ISSUE THREAD

**Issue by jondruse** [rmosolgo/graphql-ruby#2906]
**Describe the bug**

fields that start with a capital letter don't follow the same rules for being resolved by a hash

**Versions**

`graphql` 1.10.6
`ruby` 

**GraphQL schema**

```ruby
class ResultType < GraphQL::Schema::Object                                                    
  field :lowercase, String, camelize: false, null: true                                       
  field :Capital, String, camelize: false, null: true
  field :Other, String, camelize: true, null: true                                         
  field :OtherCapital, String, camelize: false, null: true, hash_key: "OtherCapital"          
end                                                                                           
                                                                                              
class QueryType < GraphQL::Schema::Object                                                     
                                                                                              
  field :search_results, [ResultType], camelize: false, null: false                           
                                                                                              
                                                                                              
  def search_results                                                                          
    [                                                                                         
      {                                                                                       
        "lowercase" => "lowercase field works",                                                     
        "Capital" => "this one doesn't work",                                                 
        "Other" => "also doesn't work",
        "OtherCapital" => "this one works"                                                    
      }                                                                                       
    ]                                                                                         
  end                                                                                         
                                                                                                                                                                                
end                                                                                           
                                                                                              
class MyAppSchema < GraphQL::Schema                                                           
  query QueryType                                                                             
end                                                                                         

 ```

**GraphQL query**
```graphql
query {
  search_results { lowercase Capital Other OtherCapital } 
}
```

```json
{
  "data": {
    "search_results": [
      {
        "lowercase": "lowercase field works",
        "Capital": null,
        "Other": null,
        "OtherCapital": "this one works"
      }
    ]
  }
}
```

**Expected behavior**

I would expect that fields starting with a capital letter would behave the same as fields that don't.


**Additional context**

My `BaseField` class specifies `camelize: false` because we're upgrading an exiting GraphQL schema that is in heavy use and we don't want to force a ton of other changes.  I understand also that fields that start with a capital letter are probably "doing it wrong" but this is what im stuck with.  The good news is that the work around is quite simple.

Thanks again for all you've done on this!

**Comment by rmosolgo** [comment]
Thanks for the bug report. I agree it should work like you said. However, it will be a bit of a breaking fix to make it work. Right now, it uses `underscored_name` (instead of `name`, as configured) to make hash keys: 

https://github.com/rmosolgo/graphql-ruby/blob/ee19fcb787106f45d0ce8a0d0b8036fc27959b7b/lib/graphql/schema/field.rb#L239-L244

Those keys (which have underscores) are later looked up on the object: 

https://github.com/rmosolgo/graphql-ruby/blob/ee19fcb787106f45d0ce8a0d0b8036fc27959b7b/lib/graphql/schema/field.rb#L608-L615


So, if they were (correctly) generated via the configured name, it would work like you expect it to. 

I'll try to get a fix into 1.11.0.

**Comment by rmosolgo** [comment]
Fixed by #2938, to be released in 1.11.0 (hopefully the end of this week!). You can give it a try from GitHub: 

```
gem "graphql", github: "rmosolgo/graphql-ruby", branch: "1.11-dev"
```
